packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -778,7 +778,7 @@ export interface CallWithChatAdapterManagement {
     deleteMessage(messageId: string): Promise<void>;
     disposeStreamView(remoteUserId?: string, options?: VideoStreamOptions): Promise<void>;
     // (undocumented)
-    downloadAuthenticatedAttachment?: (attachmentUrl: string) => Promise<AttachmentDownloadResult>;
+    downloadAuthenticatedAttachment?: (attachmentUrl: string) => Promise<AttachmentDownloadResult | undefined>;
```

Comment:
What does it means to return undefined? Like it is failed?

Reply:
At the moment the only time we return `undefined` is if `downloadAuthenticatedAttachment` is not there. Since it is an optional method.

Reply:
For example if an attachment they don't want to process for some reason, because returning `{ blobUrl: '' }` with empty string doesn't really make sense, so would return undefined instead

Reply:
We current don't make adapter method optional - but would be a good discussion point we can have for the team

Currently the definition for adapter is - that is what exactly needed by composite, and all functions there are required to get composite work nicely

For whoever implement adapter directly - that is not what we support for non-breaking change since it is a very advanced method - but they still have way to get over it, people can just implement an empty function to bypass the type check

Reply:
If you want to keep the method optional, we should still remove the `| undefined`, the case of the callwithchat adapter it should either call chatAdapter.downloadAuthenticatedAttachment or throw an error if thats not implemented